test/e2e/e2e_test.go

```
@@ -0,0 +1,154 @@
+					Labels: labels,
+				},
+				Spec: MakePodSpec(),
+			},
+		},
+	}
+}
+
+// startEndToEndForLowNodeUtilization tests the lownode utilization strategy.
+func startEndToEndForLowNodeUtilization(clientset clientset.Interface) {
+	var thresholds = make(deschedulerapi.ResourceThresholds)
+	var targetThresholds = make(deschedulerapi.ResourceThresholds)
+	thresholds[v1.ResourceMemory] = 20
+	thresholds[v1.ResourcePods] = 20
+	thresholds[v1.ResourceCPU] = 85
+	targetThresholds[v1.ResourceMemory] = 20
+	targetThresholds[v1.ResourcePods] = 20
+	targetThresholds[v1.ResourceCPU] = 90
+	// Run descheduler.
+	evictionPolicyGroupVersion, err := eutils.SupportEviction(clientset)
+	if err != nil || len(evictionPolicyGroupVersion) == 0 {
+		glog.Fatalf("%v", err)
+	}
+	stopChannel := make(chan struct{})
+	nodes, err := nodeutil.ReadyNodes(clientset, "", stopChannel)
+	if err != nil {
+		glog.Fatalf("%v", err)
+	}
+	nodeUtilizationThresholds := deschedulerapi.NodeResourceUtilizationThresholds{Thresholds: thresholds, TargetThresholds: targetThresholds}
+	nodeUtilizationStrategyParams := deschedulerapi.StrategyParameters{NodeResourceUtilizationThresholds: nodeUtilizationThresholds}
+	lowNodeUtilizationStrategy := deschedulerapi.DeschedulerStrategy{Enabled: true, Params: nodeUtilizationStrategyParams}
+	ds := &options.DeschedulerServer{Client: clientset}
+	strategies.LowNodeUtilization(ds, lowNodeUtilizationStrategy, evictionPolicyGroupVersion, nodes)
+	time.Sleep(10 * time.Second)
+
+	return
+}
+
+func TestE2E(t *testing.T) {
+	// If we have reached here, it means cluster would have been already setup and the kubeconfig file should
+	// be in /tmp directory.
+	clientSet, err := client.CreateClient("/tmp/admin.conf")
+	if err != nil {
+		t.Errorf("Error during client creation with %v", err)
+	}
+	nodeList, err := clientSet.Core().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		t.Errorf("Error listing node with %v", err)
+	}
+	leastLoadedNode := nodeList.Items[2]
```

Nit: We should check if the cluster has at least 3 nodes. Otherwise this will panic.

Btw. how do you the third node in the list is actually the least loaded node?

---

> Nit: We should check if the cluster has at least 3 nodes. Otherwise this will panic.

We would have failed in gcloud_cluster_create.sh if we don't have 3 nodes. I thought failing early is better. 


> Btw. how do you the third node in the list is actually the least loaded node?

I have added a comment here.

https://github.com/kubernetes-incubator/descheduler/pull/31/files/e0d3b2663509dde3266daed9ecd0867f6fc4fdb0#diff-6413e1dbafd6dd63b5d907ea3ae6f6f9R151

---

Ok, let's put the comment here. Smth like:
```
// Assumption: kubeadm brings all the master components onto master node. So, the last node would have least utilization.
```

---

Sure.